### ISSUE TITLE
Parse Error

### ISSUE THREAD

**Issue by CHR-IS-B** [rubberduck-vba/Rubberduck#3652]
RD shows a parse error at the line starting with "Else:".  It says "no viable alternative at input 'Else'".  The code compiles and runs.  I'm running RD 2.1.1.2532.

I can easily change the code to avoid the error so I'm just sharing this for your benefit.
```
Option Explicit
Private failCount As Long

Private Function TestGetObject(ByVal functionName As String) As Boolean
    Debug.Print "Testing '" & functionName & "'"
    On Error Resume Next
    If Not Application.Run(functionName) Is Empty Then TestGetObject = True
    On Error GoTo 0
    If TestGetObject Then: Debug.Print "  Passed": _
    Else: Debug.Print "  --FAILED--": failCount = failCount + 1
End Function
```

**Comment by rubberduck203** [comment]
Thanks for reporting this. You say you can easily modify the code so the parsing error can be avoided. Do you mind sharing the version that parses properly?

**Comment by retailcoder** [comment]
@rubberduck203 my money is on the instruction separators 😉

**Comment by rubberduck203** [comment]
Yeah. Mine is too. Just wanted to verify.

**Comment by bclothier** [comment]
Indeed. Just to borrow another repro from the chat earlier --

```
Public Sub Blah()
If True Then:
End Sub
```

We need to ensure that the colon + whitespace + new line are treated as basically whitespace. But we have to be careful with this:

```
Public Sub Blah()
If True Then: Debug.Print "It's still a single-line statement": 'Try adding `End If` here. It won't happen
End Sub
```

**Comment by CHR-IS-B** [comment]
This version parsed without error.  I dropped the colon after `Else`.
```
Option Explicit
Private failCount As Long

Private Function TestGetObject(ByVal functionName As String) As Boolean
    Debug.Print "Testing '" & functionName & "'"
    On Error Resume Next
    If Not Application.Run(functionName) Is Empty Then TestGetObject = True
    On Error GoTo 0
    If TestGetObject Then: Debug.Print "  Passed": _
    Else Debug.Print "  --FAILED--": failCount = failCount + 1
End Function
```

**Comment by bclothier** [comment]
FYI, I can no longer reproduce this using the code from the original report. Copying the code as-is into an Excel project with recent pre-release of RD, I did not get a parser error.

However, I still do get a parser error with the 2nd example:

```
Public Sub Blah()
If True Then: Debug.Print "It's still a single-line statement": 'Try adding `End If` here. It won't happen
End Sub
```

So it may have been partially fixed in the intervening time.

**Comment by MDoerner** [comment]
In which host are you getting the error?

Both after inserting the 2nd example in an Excel module and a unit test I do not get any parse error.

**Comment by bclothier** [comment]
In Excel using `ThisWorkbook` but the story is more interesting than that.

I just tested the 2nd example in isolation and I do not get a parser error. So I deleted the 2nd example and went to the 1st example. Still no parser error. But if I then add _both_ 1st and 2nd examples to the `ThisWorkbook` module, then I get a parser error....

I should note that the pasting of the 2nd example is apparently chaotic; depending on when I cut'n'paste, I may get parser error.
